frontend/src/components/Patchnotes.jsx

```diff
@@ -20,53 +34,68 @@ const Patchnotes = () => {
         };
 
         fetchPatchDetails();
-    }, [year, month]);
+    }, [game, year, month, version]);
 
     return (
         <div>
-            <h2>Tank Updates</h2>
-            {patchDetails.tank.length > 0 ? (
-                patchDetails.tank.map((content, index) => (
-                    <div key={index} dangerouslySetInnerHTML={{ __html: content }} />
-                ))
-            ) : (
-                <p>No tank updates found.</p>
-            )}
+            {game === 'overwatch' ? (
+                <>
+                    <h2>Tank Updates</h2>
+                    {patchDetails.tank.length > 0 ? (
+                        patchDetails.tank.map((content, index) => (
+                            <div key={index} dangerouslySetInnerHTML={{ __html: content }} />
+                        ))
+                    ) : (
+                        <p>No tank updates found.</p>
+                    )}
 
-            <h2>Damage Updates</h2>
-            {patchDetails.damage.length > 0 ? (
-                patchDetails.damage.map((content, index) => (
-                    <div key={index} dangerouslySetInnerHTML={{ __html: content }} />
-                ))
-            ) : (
-                <p>No damage updates found.</p>
-            )}
+                    <h2>Damage Updates</h2>
+                    {patchDetails.damage.length > 0 ? (
+                        patchDetails.damage.map((content, index) => (
+                            <div key={index} dangerouslySetInnerHTML={{ __html: content }} />
+                        ))
+                    ) : (
+                        <p>No damage updates found.</p>
+                    )}
 
-            <h2>Support Updates</h2>
-            {patchDetails.support.length > 0 ? (
-                patchDetails.support.map((content, index) => (
-                    <div key={index} dangerouslySetInnerHTML={{ __html: content }} />
-                ))
+                    <h2>Support Updates</h2>
+                    {patchDetails.support.length > 0 ? (
+                        patchDetails.support.map((content, index) => (
+                            <div key={index} dangerouslySetInnerHTML={{ __html: content }} />
+                        ))
+                    ) : (
+                        <p>No support updates found.</p>
+                    )}
+                </>
             ) : (
```

Comment:
the repeated and nested ternary operators are kind of difficult to read. I think you could improve the legibilty quite a bit by extracting some of the inner pieces into separate components

imagine something like
```
const PatchSection = ({ title, details, emptyState }) => {
  return <>
    <h2>{title}</h2>
    {details.length > 0 ? (
      details.map((detail, index) => (
        <PatchDetail detail={detail} index={index} />
      ))
    ) : (
      <p>{emptyState}</p>
    )}
  </>;
};
```

and you can use this with

```
<PatchSection title="Tank Updates" details={patchDetails.tank} emptyState="No tank updates found." />
<PatchSection title="Support Udpates" details={patchDetails.support} emptyState="No support updates found." />
```
et cetera